File: Server/DBModels/Tag.cs
using System;
using System.Collections.Generic;

namespace Server.DBModels
{
    public partial class Tag
    {
        public Tag()
        {
            UserTag = new HashSet<UserTag>();
        }

        public string Id { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string Color { get; set; }
        public string Position { get; set; }
        public string Comment { get; set; }
        public string Date { get; set; }
        public string Variable { get; set; }

        public ICollection<UserTag> UserTag { get; set; }
    }
}

File: Client/src/app/vis/Tag.ts
export class Tag {
    constructor(
        public name: string,
        public type: string,
        public color: string,
        public position: string,
        public comment: string,
        public date?: string,
        public variable?: string,
        public id?: string
    ) { }

    static Clone(tag: Tag) {
        return new Tag(
            tag.name,
            tag.type,
            tag.color,
            tag.position,
            tag.comment,
            tag.date,
            tag.variable,
            tag.id
        );
    }
}
